lib/commands/moveToFinished-3.lua

```diff
@@ -0,0 +1,56 @@
+--[[
+  Move job from active to a finished status (completed o failed)
+  A job can only be moved to completed if it was active.
+  The job must be locked before it can be moved to a finished status,
+  and the lock must be released in this script.
+
+     Input:
+      KEYS[1] active key
+      KEYS[2] completed/failed key
+      KEYS[3] jobId key
+
+      ARGV[1]  jobId
+      ARGV[2]  timestamp
+      ARGV[3]  msg property
+      ARGV[4]  return value / failed reason
+      ARGV[5]  token
+      ARGV[6]  shouldRemove
+      ARGV[7]  event channel
+      ARGV[8]  event data (? maybe just send jobid).
+
+     Output:
+      0 OK
+      -1 Missing key.
```

Comment:
you are right. I added a -2 error code when a lock is missing. I noticed however that this error codes are not being used, I need to fix it as well, it may be hidding errors right now.